src/App.tsx

```diff
@@ -1,60 +1,126 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './components/types/Todo';
+
+export function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+function getTodosId(todoId: Todo[]) {
+  const maxTodoId = Math.max(...todoId.map(todo => todo.id));
+
+  return maxTodoId + 1;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, sethasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [userIdError, setUserIdError] = useState(false);
+  const [todoList, setTodoList] = useState<Todo[]>(todos);
+  const newId = getTodosId(todoList);
+
+  const handleTitleUpdate = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    sethasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUserIdError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setUserIdError(!userId);
+
+    if (!title.trim()) {
+      sethasTitleError(true);
+    }
+
+    if (userId > 0 && title.trim() !== '') {
```

Comment:
- Don't compare with an empty string, just check if a value exists.
- `title.trim()` you do it twice, create a variable